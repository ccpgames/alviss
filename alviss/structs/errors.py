__all__ = [
    'AlvissError',
    'AlvissStubberError',
    'AlvissStubberSyntaxError',
    'AlvissStubberInvalidTypeName',
]


class AlvissError(Exception):
    pass


class AlvissStubberError(AlvissError):
    pass


class AlvissStubberSyntaxError(AlvissStubberError, ValueError):
    def __init__(self, message: str, field_name: str = '?'):
        super().__init__(message)
        self.field_name = field_name

    def __str__(self):
        return f'{super().__str__()} (field_name="{self.field_name}")'


class AlvissStubberInvalidTypeName(AlvissStubberSyntaxError, TypeError):
    def __init__(self, message: str, field_name: str = '?', type_name: str = '?'):
        super().__init__(message, field_name)
        self.type_name = type_name

    def __str__(self):
        return f'{super().__str__()} (type_name="{self.type_name}", field_name="{self.field_name}")'
