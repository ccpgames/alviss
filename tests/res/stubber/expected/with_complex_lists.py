__all__ = [
    'CfgFooStub',
    'CfgFoolStub',
    'CfgBarYomamaStub',
    'CfgBarStub',
    'CfgStub',
]

from typing import *
from alviss.structs import Empty
from alviss.structs.cfgstub import _BaseCfgStub
from alviss.structs import BaseConfig


class CfgFooStub(_BaseCfgStub, dict):
    my_list: Union[List[str], Empty]
    my_required_list: List[str]


class CfgFoolStub(_BaseCfgStub, dict):
    myDifferentList: Union[List[str], Empty]
    myDifferentMultiTypeList: Union[List[Union[str, float]], Empty]


class CfgBarYomamaStub(_BaseCfgStub, dict):
    myDifferentRequiredList: List[Union[str, int]]
    myDifferentRequiredListByKeyName: List[Any]


class CfgBarStub(_BaseCfgStub, dict):
    yomama: CfgBarYomamaStub


class CfgStub(_BaseCfgStub, dict):
    Foo: CfgFooStub
    fool: Union[CfgFoolStub, Empty]
    bar: CfgBarStub
