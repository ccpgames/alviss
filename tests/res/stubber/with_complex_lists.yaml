Foo:  # Should be required
  my_list: list[str]  # Optional
  my_required_list: List[str]*  # Should be required
fool: # Optional
  myDifferentList: # Optional
    - str
  myDifferentMultiTypeList: # Optional
    - str|float
bar: # Should be required
  yomama: # Should be required
    myDifferentRequiredList: # Should be required
      - str|int*
    myDifferentRequiredListByKeyName*: # Should be required
      - any