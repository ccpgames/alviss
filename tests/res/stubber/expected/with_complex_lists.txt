__all__ = [
    'CfgFooStub',
    'CfgFoolStub',
    'CfgBarYomamaStub',
    'CfgBarStub',
    'CfgStub',
    'AlvissConfigStub',
]

from typing import *
from alviss.structs import Empty
from alviss.structs.cfgstub import _BaseCfgStub
from alviss.structs import BaseConfig


class CfgFooStub(_BaseCfgStub):
    my_list: Union[List[str], Empty]
    my_required_list: List[str]


class CfgFoolStub(_BaseCfgStub):
    myDifferentList: Union[List[str], Empty]
    myDifferentMultiTypeList: Union[List[Union[str, float]], Empty]


class CfgBarYomamaStub(_BaseCfgStub):
    myDifferentRequiredList: List[Union[str, int]]
    myDifferentRequiredListByKeyName: List[Any]


class CfgBarStub(_BaseCfgStub):
    yomama: CfgBarYomamaStub


class CfgStub(_BaseCfgStub):
    Foo: CfgFooStub
    fool: Union[CfgFoolStub, Empty]
    bar: CfgBarStub


class AlvissConfigStub(BaseConfig, CfgStub):
    pass
